packages/frontend/src/components/Trade/Short/OpenShort.tsx

```diff
@@ -0,0 +1,732 @@
+import { BIG_ZERO, MIN_COLLATERAL_AMOUNT } from '@constants/index'
+import {
+  connectedWalletAtom,
+  isTransactionFirstStepAtom,
+  networkIdAtom,
+  supportedNetworkAtom,
+} from 'src/state/wallet/atoms'
+import { addressesAtom, isLongAtom, vaultHistoryUpdatingAtom } from 'src/state/positions/atoms'
+import { useComputeSwaps, useFirstValidVault } from 'src/state/positions/hooks'
+import { useVaultData } from '@hooks/useVaultData'
+import { useVaultManager } from '@hooks/contracts/useVaultManager'
+import {
+  useGetCollatRatioAndLiqPrice,
+  useGetDebtAmount,
+  useGetUniNFTCollatDetail,
+  useUpdateOperator,
+} from 'src/state/controller/hooks'
+import UniswapData from '../UniswapData'
+import Confirmed, { ConfirmType } from '@components/Trade/Confirmed'
+import Cancelled from '../Cancelled'
+import useAppEffect from '@hooks/useAppEffect'
+import useAppCallback from '@hooks/useAppCallback'
+import useAppMemo from '@hooks/useAppMemo'
+import { useVaultHistoryQuery } from '@hooks/useVaultHistory'
+import { currentImpliedFundingAtom, dailyHistoricalFundingAtom } from 'src/state/controller/atoms'
+import VaultCard from './VaultCard'
+import ConfirmApproval from './ConfirmApproval'
+import TradeDetails from '../TradeDetails'
+import { useETHPrice } from '@hooks/useETHPrice'
+import useDebounce from '@utils/useDebounce'
+import { Networks } from '../../../types'
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    buttonDiv: {
+      position: 'sticky',
+      bottom: '0',
+      background: '#2A2D2E',
+      paddingBottom: theme.spacing(3),
+    },
+    settingsButton: {
+      marginTop: theme.spacing(2),
+    },
+    settingsContainer: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      width: '300px',
+      margin: '0 auto',
+    },
+    explainer: {
+      marginTop: theme.spacing(2),
+      paddingRight: theme.spacing(1),
+    },
+    thirdHeading: {
+      marginTop: theme.spacing(2),
+    },
+    formHelperText: {
+      marginLeft: 0,
+      marginRight: 0,
+    },
+    amountInput: {
+      marginTop: theme.spacing(1),
+      backgroundColor: `${theme.palette.error.main}aa`,
+      '&:hover': {
+        backgroundColor: theme.palette.error.dark,
+      },
+    },
+    caption: {
+      marginTop: theme.spacing(1),
+      marginBottom: theme.spacing(1),
+      fontSize: '13px',
+    },
+    hint: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    arrowIcon: {
+      marginLeft: '4px',
+      marginRight: '4px',
+      fontSize: '20px',
+    },
+    hintTextContainer: {
+      display: 'flex',
+    },
+    hintTitleText: {
+      marginRight: '.5em',
+    },
+    divider: {
+      margin: theme.spacing(2, 0),
+      width: '300px',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+    },
+  }),
+)
+
+const FUNDING_MOVE_THRESHOLD = 0.7
+
+export const OpenShortPosition = () => {
+  const classes = useStyles()
+  const [confirmedAmount, setConfirmedAmount] = useState('0')
```

Comment:
I am concerned with the amount states variables we are using. The trade page flickers a lot

Reply:
How can we make this better?

Reply:
Hey @KMKoushik, I'm not sure what you mean when you say it's very hard to change the collat ratio. Currently, there is a minimum collat ratio value, below which the user cannot set their collat ratio.

Reply:
@KMKoushik about the state variables, I have removed three of them that I thought were redundant.

Reply:
I made PR up for fixing the flickering issue @KMKoushik mentioned. After playing his video slowly, I realized that flickering is related to the vault that is sometimes defined but not defined sometimes on his local. Its changes are too fast, so we see flickering. I can't reproduce the same thing on my local but I think it's related to a performance issue in the vault manager.
https://github.com/opynfinance/squeeth-monorepo/pull/560